views/js/runner/plugins/tools/calculator.js

```diff
@@ -0,0 +1,185 @@
+                if(context.options.calculator){//allow calculator
+                    self.show();
+                }else{
+                    self.hide();
+                }
+            }
+
+            //build element (detached)
+            this.$button = $(buttonTpl({
+                control : 'calculator',
+                title : __('Open Calculator'),
+                icon : 'maths',
+                text : __('Calculator')
+            }));
+            this.$calculatorContainer = $('<div class="widget-calculator">');
+            
+            //init calculator instance var, it will be created only necessary
+            this.calculator = null;
+
+            //attach behavior
+            this.$button.on('click', function (e){
+                
+                //get the offset of the button to position the calculator widget close to it
+                var offset = $(this).offset();
+
+                //prevent action if the click is made inside the form which is a sub part of the button
+                if($(e.target).closest('.widget-container').length){
+                    return;
+                }
+
+                e.preventDefault();
+
+                if(self.getState('enabled') !== false){
+                    if(self.calculator){
+                        //just show/hide the calculator widget
+                        if(self.calculator.is('hidden')){
+                            self.calculator.show();
+                        }else{
+                            self.calculator.hide();
+                        }
+                    }else{
+                        //build calculator widget
+                        self.calculator = calculatorFactory({
+                            renderTo : self.$calculatorContainer,
+                            replace : true,
+                            draggableContainer : areaBroker.getContainer().find('.test-runner-sections')[0],
+                            width : _default.width,
+                            height : _default.height,
+                            top : offset.top - _default.height - 40,
+                            left : offset.left
+                        });
+                    }
+                }
+            });
+
+            //start disabled
+            togglePlugin();
+            this.disable();
+
+            //update plugin state based on changes
+            testRunner
+                .on('loaditem', togglePlugin)
+                .on('renderitem', function (){
+                    self.enable();
+                })
+                .on('unloaditem', function (){
+                    self.disable();
+                    if(self.calculator){
+                        //destroy calculator to create a new instance of calculator each time
+                        self.calculator.destroy();
+                        self.calculator = null;
+                    }
+                });
+
+            areaBroker.getToolboxArea().append(this.$button);
```

Comment:
Could you move everything above into the `init` method. The goal is to make the render step the shortest (only append to the dom), when element creation, event binding, etc. are supposed to be made during the initialization